### PR TITLE
Scale fonts and padding by viewport size

### LINKED ISSUE #4: Responsive view
The current view looks horrible on small devices. It could be great to have an interface that looks reasonable well on mobile / small screen devices.

### PR DESCRIPTION
**Pull request by JimBledsoe**
Fixes #4

### CODE REVIEW

**Comment by JimBledsoe** [comment]
There are far more changes than I anticipated. I will annotate the PR.

File: src/index.js [hunk 1]
```diff
@@ -178,6 +178,12 @@ class IPAddress extends Component {
             <span className="value">{details.size.toLocaleString()}</span>
             <span className="label">Count</span>
           </span>
+        </div>
+
+        <div className="break">
+        </div>
+
+        <div className="usable">
```
Comment (JimBledsoe) [inline src/index.js:186]: Thit virtual div is to ensure the first and last usable IP's are always on their own row. I noticed working on this that the footer has the explanation about the base and broadcast. It made more sense to me to just force this to be two lines and keep the first and last near the footer footnote.

File: src/style.scss [hunk 2]
```diff
@@ -14,16 +20,17 @@ body {
     margin-bottom: 30px;
     margin-left: 40px;
     h1 {
-      font-size: modular-scale(2);
       small {
```
Comment (JimBledsoe) [inline src/style.scss:23]: I could not see there the module-scale was working at all. Perhaps it was installed at some point and then removed.

File: src/style.scss [hunk 3]
```diff
@@ -14,16 +20,17 @@ body {
...
       }
     }
     p {
-      margin-right: 90px;
-      font-size: modular-scale(-1);
+      margin-right: 30px;
+      @media ( min-width: $med ) {
+        margin-right: 60px;
+      }
     }
```
Comment (JimBledsoe) [inline src/style.scss:34]: This looked better as things got tight.

File: src/style.scss [hunk 4]
```diff
@@ -37,15 +44,21 @@ body {
       input {
         width: 2.5em;
         font-size: 2.7em;
+        @media (max-width: $xl) {
+          font-size: calc(1em + 1.9vw);
+        }
         text-align: center;
```
Comment (JimBledsoe) [inline src/style.scss:50]: This is the general pattern on fonts. Set a max that was nearly identical to your original and don't go any larger on really large displays, but get smaller on smaller displays.

File: src/style.scss [hunk 5]
```diff
@@ -98,47 +114,54 @@ body {
...
-      span.count {
-        display: inline-block;
-        width: 15%;
+      span {
+        display: block;
       }
-      span.first {
-        display: inline-block;
-        width: 15%;
+    }
```
Comment (JimBledsoe) [inline src/style.scss:126]: Break is just a dummy flex element full width and no height to force a row break.

File: src/style.scss [hunk 6]
```diff
@@ -98,47 +114,54 @@ body {
...
+        width: 640px;
       }
-      span.last {
-        display: inline-block;
-        width: 15%;
+      span {
+        display: block;
       }
-      span.value {
-        font-size: modular-scale(1);
```
Comment (JimBledsoe) [inline src/style.scss:126]: Value and label just moved out to stand on their own for re-use.

File: src/style.scss [hunk 7]
```diff
@@ -98,47 +114,54 @@ body {
...
+      font-size: 1.22em;
+      @media (max-width: $xl) {
+        font-size: calc(0.7em + 0.6vw);
       }
+      font-weight: bold;
+      color: #656582;
     }
   }
 
   footer {
```
Comment (JimBledsoe) [inline src/style.scss:163]: Move the footer in tighter since it really corresponds strongly to the last row of flex data.

File: src/style.scss [hunk 8]
```diff
@@ -98,47 +114,54 @@ body {
       }
     }
     div.details {
```
Comment (JimBledsoe) [inline src/style.scss:116]: Changed to a nested combo of a flex of rows and block within the row (value over label).
Moved common props out to parent div or span in most cases.
Scale fonts slightly, keeping a max set.
Max of 640 is to keep a pleasing set of 4x or 2x flex units so they don't scale too far apart, but let them get as tight as they need to.

**Comment by JimBledsoe** [comment]
Same URL as I emailed directly to you @yuvadm a few PR's ago to test live site.

**Comment by JimBledsoe** [comment]
image

**Comment by JimBledsoe** [comment]
@yuvadm if you are not interested in this, just let me know and I will close it.

**Review by aravindvnair99 — APPROVED** [review]

**Pull request merged**